src/shared/ptyfwd.c

```diff
@@ -1083,26 +1110,29 @@ PTYForward* pty_forward_free(PTYForward *f) {
 }
 
 int pty_forward_set_ignore_vhangup(PTYForward *f, bool b) {
-        int r;
+        bool changed = false;
 
         assert(f);
 
-        if (FLAGS_SET(f->flags, PTY_FORWARD_IGNORE_VHANGUP) == b)
-                return 0;
-
-        SET_FLAG(f->flags, PTY_FORWARD_IGNORE_VHANGUP, b);
+        if (FLAGS_SET(f->flags, PTY_FORWARD_IGNORE_VHANGUP) != b) {
+                SET_FLAG(f->flags, PTY_FORWARD_IGNORE_VHANGUP, b);
+                changed = true;
+        }
 
-        if (!ignore_vhangup(f)) {
+        /* When false, also unset the initial one. */
+        if (!b && FLAGS_SET(f->flags, PTY_FORWARD_IGNORE_INITIAL_VHANGUP)) {
```

Comment:
Hmm, not sure I follow? It's perfectly reasonable to only ignore the initial one while resetting the generic one to false?

Reply:
To make it usable in systemd-run, which uses PTY_FORWARD_IGNORE_INITIAL_VHANGUP.

Alternatively, I think it is OK to introduce, e.g. pty_forward_set_ignore_initial_vhangup() or pty_forward_set_flags().

Reply:
That sounds better!